### PR TITLE
chore(client): use Spacer component from @freecodecamp/ui

### LINKED ISSUE #56744: Replace Spacer with the one from @freecodecamp/ui
## Description

We have implemented a `Spacer` component in our component library (`@freecodecamp/ui`), so the existing `Spacer` in the /learn client should be replaced with that:

```diff
--- import { Loader, Spacer } from '../components/helpers';
+++ import { Spacer } from '@freecodecamp/ui';
```

Note that this is not a direct swap. The `size` prop needs to be updated as follows:

| /learn Spacer | @freecodecamp/ui Spacer |
| --- | --- |
| `xxSmall` | `xxs` |
| `small` | `xs` |
| `xSmall` | `s` |
| `medium` | `m` |
| `large` | `l` |
| `exLarge` | `xl` |
| `doubleXL` | `xxl` |

## Changes needed
 
- Migrate all Spacer usages to `@freecodecamp/ui` following the mapping above
- Remove the Spacer component from the codebase: https://github.com/freeCodeCamp/freeCodeCamp/blob/main/client/src/components/helpers/spacer.tsx

### PR DESCRIPTION
**Pull request by RGHANILOO**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #56744

### CODE REVIEW

**Review by huyenltnguyen — COMMENTED** [review]
Thank you for the PR.

The second requirement (mentioned in the issue description) still needs to be addressed.

File: client/src/components/helpers/index.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
 export { default as FullWidthRow } from './full-width-row';
 export { default as Loader } from './loader';
 export { default as SkeletonSprite } from './skeleton-sprite';
-export { default as Spacer } from './spacer';
+export { default as Spacer } from '@freecodecamp/ui';
```
[the remark this answers is not in the document]
Comment (RGHANILOO) [reply client/src/components/helpers/index.ts]: this has now been resolved.

**Comment by ilenia-magoni** [comment]
the Linter is failing on 7 different files, please fix that.

**Comment by ilenia-magoni** [comment]
I added a commit that solved the lint errors, now there are the typescript errors to solve

**Pull request closed without merging**